lib/view_component/engine.rb

```diff
@@ -110,11 +110,15 @@ class Engine < Rails::Engine # :nodoc:
     end
 
     initializer "static assets" do |app|
-      if app.config.view_component.show_previews
+      if serve_static_previews? app.config
         app.middleware.use(::ActionDispatch::Static, "#{root}/app/assets/vendor")
       end
     end
 
+    def serve_static_previews?(app_config)
```

Comment:
Can this be extracted into a module perhaps?

```suggestion
    def serve_static_preview_assets?(app_config)
```